presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/nativechecker/NativePlanChecker.java

```diff
@@ -70,12 +81,31 @@ public void validate(PlanNode planNode, WarningCollector warningCollector, Conne
     @Override
     public void validateFragment(SimplePlanFragment planFragment, WarningCollector warningCollector, ConnectorSession session)
     {
-        if (!planFragment.getPartitioning().isCoordinatorOnly() && !isInternalSystemConnector(planFragment.getRoot())) {
-            runValidation(planFragment);
-        }
-        else {
+        if (planFragment.getPartitioning().isCoordinatorOnly()
+                || isInternalSystemConnector(planFragment.getRoot())) {
             LOG.debug("Skipping native plan validation [fragment: %s, root: %s]", planFragment.getId(), planFragment.getRoot().getId());
+            return;
         }
+        runValidation(removeTableWriter(planFragment));
+    }
+
+    private SimplePlanFragment removeTableWriter(SimplePlanFragment planFragment)
```

Comment:
It might be good to add a note as to why the TableWriterNode gets replaced

Reply:
+1.